tests/unit/fs_mock_metadata_to_disk_buffer_test.cc

```diff
@@ -0,0 +1,398 @@
+    BOOST_REQUIRE_EQUAL(elem.header.dir_inode, add_dir_entry_op.dir_inode);
+    BOOST_REQUIRE_EQUAL(elem.header.entry_inode, add_dir_entry_op.entry_inode);
+    BOOST_REQUIRE_EQUAL(elem.header.entry_name_length, add_dir_entry_op.entry_name_length);
+    BOOST_REQUIRE_EQUAL(std::string(elem.entry_name.get(), elem.entry_name.end()), add_dir_entry_str);
+}
+
+SEASTAR_THREAD_TEST_CASE(create_inode_as_dir_entry_test) {
+    mock_metadata_to_disk_buffer buf(default_buff_size, default_alignment);
+    buf.init(0);
+
+    std::string create_inode_str = "120345";
+    ondisk_create_inode_as_dir_entry_header create_inode_op {
+            {42, 1, {5, 2, 6, 8, 4}}, 7, static_cast<uint16_t>(create_inode_str.size())
+        };
+    BOOST_REQUIRE_EQUAL(buf.append(create_inode_op, create_inode_str.c_str()), APPENDED);
+
+    BOOST_REQUIRE_EQUAL(buf.bytes_left(), default_buff_size - checkpoint_size -
+            get_ondisk_entry_size(create_inode_op));
+
+    BOOST_CHECK(buf.is_append_type<ondisk_create_inode_as_dir_entry>(0));
+    auto &elem = buf.get_by_append_type<ondisk_create_inode_as_dir_entry>(0);
+    BOOST_CHECK(std::memcmp(&elem.header.entry_inode, &create_inode_op.entry_inode,
+            sizeof(elem.header.entry_inode)) == 0);
+    BOOST_REQUIRE_EQUAL(elem.header.dir_inode, create_inode_op.dir_inode);
+    BOOST_REQUIRE_EQUAL(elem.header.entry_name_length, create_inode_op.entry_name_length);
+    BOOST_REQUIRE_EQUAL(std::string(elem.entry_name.get(), elem.entry_name.end()), create_inode_str);
+}
+
+SEASTAR_THREAD_TEST_CASE(delete_dir_entry_test) {
+    mock_metadata_to_disk_buffer buf(default_buff_size, default_alignment);
+    buf.init(0);
+
+    std::string delete_entry_str = "120345";
+    ondisk_delete_dir_entry_header delete_entry_op {42, static_cast<uint16_t>(delete_entry_str.size())};
+    BOOST_REQUIRE_EQUAL(buf.append(delete_entry_op, delete_entry_str.c_str()), APPENDED);
+
+    BOOST_REQUIRE_EQUAL(buf.bytes_left(), default_buff_size - checkpoint_size -
+            get_ondisk_entry_size(delete_entry_op));
+
+    BOOST_CHECK(buf.is_append_type<ondisk_delete_dir_entry>(0));
+    auto &elem = buf.get_by_append_type<ondisk_delete_dir_entry>(0);
+    BOOST_REQUIRE_EQUAL(elem.header.dir_inode, delete_entry_op.dir_inode);
+    BOOST_REQUIRE_EQUAL(elem.header.entry_name_length, delete_entry_op.entry_name_length);
+    BOOST_REQUIRE_EQUAL(std::string(elem.entry_name.get(), elem.entry_name.end()), delete_entry_str);
+}
+
+SEASTAR_THREAD_TEST_CASE(rename_dir_entry_test) {
+    mock_metadata_to_disk_buffer buf(default_buff_size, default_alignment);
+    buf.init(0);
+
+    std::string old_name_str = "120345";
+    std::string new_name_str = "541631";
+    ondisk_rename_dir_entry_header rename_op {
+        42,
+        24,
+        static_cast<uint16_t>(old_name_str.size()),
+        static_cast<uint16_t>(new_name_str.size())};
+    BOOST_REQUIRE_EQUAL(buf.append(rename_op, old_name_str.c_str(), new_name_str.c_str()), APPENDED);
```

Comment:
I think data() is better suited here (bonus: they are equivalent in semantics).
```suggestion
    BOOST_REQUIRE_EQUAL(buf.append(rename_op, old_name_str.data(), new_name_str.data()), APPENDED);
```
If you accept it, change it also in the above tests.